src/ui/foundation/types.ts

```diff
@@ -0,0 +1,23 @@
+import { Command } from './SettingsProvider/types';
+
+export class CommandEventTarget extends EventTarget {
```

Comment:
I would wrap EventTarget instead of extending it. (just create a class that has an instance of an eventtarget)

```
class CommandEventTarget {
   constructor() {
       this.eventtarget = new EventTarget();
  }
  on(...) {
    this.eventtarget.on(...);
  }
  execute(...) {...}
  
}
```

We don't want users of the class to call `dispatchEvent`. We want them to call `execute()`.

So we want `dispatchEvent` to be private.